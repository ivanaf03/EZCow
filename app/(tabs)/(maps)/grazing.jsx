import React from "react";

import { View, SafeAreaView } from "react-native";
import { router } from "expo-router";

import CustomButton from "../../../components/basic/custom-button";
import TabTitle from "../../../components/tabs/tab-title";

const Grazing = () => {
  return (
    <SafeAreaView className="flex-1 bg-c_dark_gray">
      <TabTitle text="Ver pastoreo" />

      <View className="mx-4 mt-4 space-y-2">
        <View>
          <CustomButton
            text="AÃ±adir al pastoreo"
            onPress={() => router.replace("grazing-form")}
            buttonTestID="grazing-form-button"
          />
        </View>
        <View className="flex-row items-center">
          <View className="w-[50%] pr-2">
            <CustomButton
              text="Ver ubicaciones"
              onPress={() => router.replace("fields")}
              buttonTestID={"fields-button"}
            />
          </View>
          <View className="w-[50%] pl-2">
            <CustomButton
              text="Ver parcelas"
              onPress={() => router.replace("cadastral-fields")}
              buttonTestID={"cadastral-fields-button"}
            />
          </View>
        </View>
      </View>
    </SafeAreaView>
  );
};

export default Grazing;
