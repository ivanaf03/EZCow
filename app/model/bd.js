import * as SQLite from "expo-sqlite";

let db;

export const getDatabase = async () => {
  if (!db) {
    db = await SQLite.openDatabaseAsync("ezcow.db");
  }
  return db;
};

export const initDatabase = async () => {
  db = await getDatabase();

  await db.execAsync(`
        PRAGMA journal_mode = WAL;

        CREATE TABLE IF NOT EXISTS User (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            email TEXT NOT NULL UNIQUE,
            password TEXT DEFAULT NULL,
            google_id TEXT DEFAULT NULL,
            latitude REAL DEFAULT 43.0,
            longitude REAL DEFAULT -7.0,
            CHECK (password IS NOT NULL OR google_id IS NOT NULL)
        );

        CREATE TABLE IF NOT EXISTS Cow (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            code TEXT NOT NULL UNIQUE,
            name TEXT NOT NULL,
            entryDate DATE NOT NULL,
            exitDate DATE DEFAULT NULL,
            gender TEXT NOT NULL,
            breed TEXT NOT NULL,
            phase TEXT NOT NULL,
            user_fk INTEGER NOT NULL,
            mother_fk INTEGER DEFAULT NULL,
            FOREIGN KEY(user_fk) REFERENCES User(id),
            FOREIGN KEY(mother_fk) REFERENCES Cow(id)
        );

        CREATE TABLE IF NOT EXISTS Field (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            latitude REAL DEFAULT 43.0,
            longitude REAL DEFAULT -7.0,
            user_fk INTEGER NOT NULL,
            FOREIGN KEY(user_fk) REFERENCES User(id)
        );

        CREATE TABLE IF NOT EXISTS BreedingEvent (
            if INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            description TEXT,
            date DATE NOT NULL,
            cow_fk INTEGER NOT NULL,
            FOREIGN KEY(cow_fk) REFERENCES Cow(id)
        );
    `);
};

export const deleteDatabase = async () => {
  await db.closeAsync();
  await SQLite.deleteDatabaseAsync("ezcow.db");
};

export const closeDatabase = async () => {
  await db.closeAsync();
  db = null;
};

export const syncDatabase = async () => {
  await db.execAsync(`
        PRAGMA wal_checkpoint(FULL);
    `);
};
