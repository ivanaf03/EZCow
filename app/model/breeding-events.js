import { getDatabase } from "./bd";

export const insertBreedingEvent = async (
  cow_fk,
  description,
  date,
  user_fk
) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `INSERT INTO BreedingEvent (cow_fk, description, date, user_fk) VALUES (?, ?, ?, ?)`,
      [cow_fk, description, date, user_fk]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getBreedingEventsByDayAnUserId = async (day, userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM BreedingEvent JOIN Cow ON BreedingEvent.cow_fk = Cow.id WHERE date = ? AND user_fk = ?`,
      [userId, day]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};