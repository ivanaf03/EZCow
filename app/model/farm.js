import { getDatabase } from "./bd";

export const insertFarmUbication = async (userId, latitude, longitude) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `UPDATE User SET latitude = ?, longitude = ? WHERE id = ?`,
      [latitude, longitude, userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getFarmUbicationByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(
      `SELECT latitude, longitude FROM User WHERE id = ?`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
