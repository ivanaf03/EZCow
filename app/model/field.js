import { getDatabase } from "./bd";

export const insertField = async (name, latitude, longitude, user_fk) => {
  const db = await getDatabase();
  try {
    await db.runAsync(
      `INSERT INTO Field (name, latitude, longitude, user_fk) VALUES (?, ?, ?, ?)`,
      [name, latitude, longitude, user_fk]
    );
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllFieldsByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(`SELECT * FROM Field WHERE user_fk = ?`, [
      userId,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
