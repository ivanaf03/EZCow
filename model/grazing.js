import { getDatabase } from "./bd";

export const insertGroup = async (name) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(`INSERT INTO CowGroup (name) VALUES (?)`, [
      name,
    ]);
    return res.lastID;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllGroups = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT DISTINCT CowGroup.id AS groupId,CowGroup.name AS groupName
       FROM CowGroup JOIN Cow ON CowGroup.id = Cow.group_fk WHERE Cow.user_fk = ? AND CowGroup.exitDate IS NULL`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const setExitDateByGroupId = async (groupId) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `UPDATE CowGroup SET exitDate = CURRENT_DATE WHERE id = ?`,
      [groupId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const insertCowInGroup = async (cowId, groupId) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(`UPDATE Cow SET group_fk = ? WHERE id = ?`, [
      groupId,
      cowId,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getCowsInGroup = async (groupId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT Cow.id AS cowId,Cow.name AS cowName,Cow.code AS cowCode,Cow.group_fk AS groupId
       FROM Cow WHERE group_fk = ?`,
      [groupId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const insertGrazing = async (fieldId, groupId) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(`INSERT INTO Grazing (field_fk, group_fk, initialDate) VALUES (?, ?, CURRENT_DATE)`, [
      fieldId,
      groupId,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getFieldIdByName = async (name) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(
      `SELECT id FROM Field WHERE name = ?`,
      [name]
    );
    return res.id;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getFieldNames = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT name FROM Field WHERE user_fk = ?`,
      [userId]
    );
    return res.map((field) => field.name);
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getGroupIdByName = async (name) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(
      `SELECT id FROM CowGroup WHERE name = ?`,
      [name]
    );
    return res.id;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getGroupNames = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT DISTINCT CowGroup.name AS groupName FROM CowGroup JOIN Cow ON CowGroup.id = Cow.group_fk WHERE Cow.user_fk = ? AND CowGroup.exitDate IS NULL`,
      [userId]
    );
    return res.map((group) => group.groupName);
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getGrazingByuserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT Grazing.id AS grazingId,Grazing.initialDate AS initialDate,Grazing.endDate AS endDate,Field.name AS fieldName,CowGroup.name AS groupName FROM Grazing JOIN Field ON Grazing.field_fk = Field.id JOIN CowGroup ON Grazing.group_fk = CowGroup.id WHERE Field.user_fk = ? AND Grazing.endDate IS NULL`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getExitedGrazingByuserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT Grazing.id AS grazingId,Grazing.initialDate AS initialDate,Grazing.endDate AS endDate,Field.name AS fieldName,CowGroup.name AS groupName FROM Grazing JOIN Field ON Grazing.field_fk = Field.id JOIN CowGroup ON Grazing.group_fk = CowGroup.id WHERE Field.user_fk = ? AND Grazing.endDate IS NOT NULL`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const expireGrazing = async (grazingId) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(`UPDATE Grazing SET endDate = CURRENT_DATE WHERE id = ?`, [
      grazingId,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};