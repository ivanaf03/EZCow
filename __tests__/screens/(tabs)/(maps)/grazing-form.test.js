import React from "react";

import { render, waitFor, fireEvent } from "@testing-library/react-native";
import { router } from "expo-router";
import { Alert } from "react-native";

import GrazingForm from "../../../../app/(tabs)/(maps)/grazing-form";
import { getGroupNames, getFieldNames, getGroupIdByName } from "../../../../model/grazing";

jest.mock("../../../../store/user-provider", () => ({
  useUser: jest.fn(() => ({
    user: { id: "1", name: "TestUser", email: "test@test.com" },
  })),
}));

jest.mock("../../../../model/grazing", () => ({
  getGroupNames: jest.fn(),
  getFieldNames: jest.fn(),
  getGroupIdByName: jest.fn(),
  getFieldIdByName: jest.fn(),
  insertGrazing: jest.fn(),
}));

jest.mock("expo-router", () => ({
  router: {
    replace: jest.fn(),
  },
}));

jest.mock("@fortawesome/react-native-fontawesome", () => ({
  FontAwesomeIcon: () => null,
}));

jest.spyOn(Alert, "alert");

describe("GrazingForm", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", async () => {
    getGroupNames.mockResolvedValue(["TestGroup"]);
    getFieldNames.mockResolvedValue(["TestField"]);
    const tree = render(<GrazingForm />);
    await waitFor(() => expect(tree).toMatchSnapshot());
  });

  it("should navigate to grazing", async () => {
    getGroupNames.mockResolvedValue(["TestGroup"]);
    getFieldNames.mockResolvedValue(["TestField"]);
    const tree = render(<GrazingForm />);
    await waitFor(() => expect(tree).toMatchSnapshot());
    const grazingButton = tree.getByTestId("grazing-button");
    fireEvent.press(grazingButton);
    await waitFor(() => expect(router.replace).toHaveBeenCalledWith("grazing"));
  });

  it("should show an error if group is empty", async () => {
    getGroupNames.mockResolvedValue([]);
    getFieldNames.mockResolvedValue(["TestField"]);
    const { getByTestId } = render(<GrazingForm />);
    await waitFor(() => expect(getGroupNames).toHaveBeenCalled());
    fireEvent(getByText("Fincas:"), "onValueChange", "TestField");
    const handleAddGrazingButton = getByTestId("handle-add-grazing-button");
    fireEvent.press(handleAddGrazingButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Por favor, rellena todos los campos."
    );
  });

  it("should show an error if field is empty", async () => {
    getGroupNames.mockResolvedValue(["TestGroup"]);
    getFieldNames.mockResolvedValue([]);
    const { getByTestId, getByText } = render(<GrazingForm />);
    await waitFor(() => expect(getFieldNames).toHaveBeenCalled());
    fireEvent(getByText("Grupos:"), "onValueChange", "TestGroup");
    const handleAddGrazingButton = getByTestId("handle-add-grazing-button");
    fireEvent.press(handleAddGrazingButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Por favor, rellena todos los campos."
    );
  });

  it("should work with valid submission", async () => {
    getGroupNames.mockResolvedValue(["TestGroup"]);
    getFieldNames.mockResolvedValue(["TestField"]);
    const { getByTestId, getByText } = render(<GrazingForm />);
    await waitFor(() => expect(getFieldNames).toHaveBeenCalled());
    fireEvent(getByText("Grupos:"), "onValueChange", "TestGroup");
    fireEvent(getByText("Fincas:"), "onValueChange", "TestField");
    const handleAddGrazingButton = getByTestId("handle-add-grazing-button");
    fireEvent.press(handleAddGrazingButton);
    await waitFor(() => {
      expect(router.replace).toHaveBeenCalledWith("grazing");
    });
  });

});