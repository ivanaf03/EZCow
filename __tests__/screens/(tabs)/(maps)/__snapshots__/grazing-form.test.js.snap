// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GrazingForm should navigate to grazing 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#1c1c1c",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 36,
          "lineHeight": 40,
        },
        {
          "fontFamily": "Nunito-Bold",
        },
        {
          "borderBottomWidth": 2,
        },
        {
          "borderBottomColor": "#ffffff",
          "borderLeftColor": "#ffffff",
          "borderRightColor": "#ffffff",
          "borderTopColor": "#ffffff",
        },
      ]
    }
  >
    Crear grupo
  </Text>
  <View
    style={
      [
        {
          "alignItems": "center",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "marginLeft": 16,
            "marginRight": 16,
          },
          {
            "marginTop": 16,
          },
          {
            "width": "75%",
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "space-between",
            },
            {
              "alignItems": "center",
            },
            {
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            },
            {
              "backgroundColor": "#272727",
            },
            {
              "borderBottomWidth": 2,
              "borderLeftWidth": 2,
              "borderRightWidth": 2,
              "borderTopWidth": 2,
            },
            {
              "borderBottomColor": "#6bc0eb",
              "borderLeftColor": "#6bc0eb",
              "borderRightColor": "#6bc0eb",
              "borderTopColor": "#6bc0eb",
            },
            {
              "borderBottomLeftRadius": 24,
              "borderBottomRightRadius": 24,
              "borderTopLeftRadius": 24,
              "borderTopRightRadius": 24,
            },
          ]
        }
        testID="grazing-button"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Nunito-SemiBold",
              },
              {
                "color": "#ffffff",
              },
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
              {
                "textAlign": "center",
              },
            ]
          }
        >
          Ver pastoreo
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 8,
          "marginRight": 8,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "display": "flex",
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "backgroundColor": "#272727",
          },
          {
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
          },
        ]
      }
    >
      <View>
        <View
          style={
            [
              {
                "marginTop": -4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontFamily": "Nunito-Light",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            Grupos
            :
          </Text>
          <View
            style={
              [
                {
                  "backgroundColor": "#ffffff",
                },
                {
                  "borderBottomWidth": 2,
                  "borderLeftWidth": 2,
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                },
                {
                  "borderBottomColor": "#2254ba",
                  "borderLeftColor": "#2254ba",
                  "borderRightColor": "#2254ba",
                  "borderTopColor": "#2254ba",
                },
                {
                  "borderBottomLeftRadius": 24,
                  "borderBottomRightRadius": 24,
                  "borderTopLeftRadius": 24,
                  "borderTopRightRadius": 24,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            <View>
              <RNCPicker
                items={[]}
                numberOfLines={1}
                onChange={[Function]}
                selectedIndex={0}
                style={
                  [
                    {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
                testID="group-picker"
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "marginTop": 24,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "marginTop": -4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontFamily": "Nunito-Light",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            Fincas
            :
          </Text>
          <View
            style={
              [
                {
                  "backgroundColor": "#ffffff",
                },
                {
                  "borderBottomWidth": 2,
                  "borderLeftWidth": 2,
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                },
                {
                  "borderBottomColor": "#2254ba",
                  "borderLeftColor": "#2254ba",
                  "borderRightColor": "#2254ba",
                  "borderTopColor": "#2254ba",
                },
                {
                  "borderBottomLeftRadius": 24,
                  "borderBottomRightRadius": 24,
                  "borderTopLeftRadius": 24,
                  "borderTopRightRadius": 24,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            <View>
              <RNCPicker
                items={[]}
                numberOfLines={1}
                onChange={[Function]}
                selectedIndex={0}
                style={
                  [
                    {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
                testID="field-picker"
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "marginBottom": 32,
              "marginTop": 32,
            },
            [
              {
                "marginTop": 24,
              },
            ],
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "display": "flex",
              },
              {
                "flexDirection": "row",
              },
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
              {
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              },
              {
                "backgroundColor": "#272727",
              },
              {
                "borderBottomWidth": 2,
                "borderLeftWidth": 2,
                "borderRightWidth": 2,
                "borderTopWidth": 2,
              },
              {
                "borderBottomColor": "#6bc0eb",
                "borderLeftColor": "#6bc0eb",
                "borderRightColor": "#6bc0eb",
                "borderTopColor": "#6bc0eb",
              },
              {
                "borderBottomLeftRadius": 24,
                "borderBottomRightRadius": 24,
                "borderTopLeftRadius": 24,
                "borderTopRightRadius": 24,
              },
            ]
          }
          testID="handle-add-grazing-button"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Nunito-SemiBold",
                },
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
                [
                  {
                    "marginLeft": 16,
                  },
                ],
              ]
            }
          >
            Añadir
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`GrazingForm should render correctly 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#1c1c1c",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 36,
          "lineHeight": 40,
        },
        {
          "fontFamily": "Nunito-Bold",
        },
        {
          "borderBottomWidth": 2,
        },
        {
          "borderBottomColor": "#ffffff",
          "borderLeftColor": "#ffffff",
          "borderRightColor": "#ffffff",
          "borderTopColor": "#ffffff",
        },
      ]
    }
  >
    Crear grupo
  </Text>
  <View
    style={
      [
        {
          "alignItems": "center",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "marginLeft": 16,
            "marginRight": 16,
          },
          {
            "marginTop": 16,
          },
          {
            "width": "75%",
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "space-between",
            },
            {
              "alignItems": "center",
            },
            {
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            },
            {
              "backgroundColor": "#272727",
            },
            {
              "borderBottomWidth": 2,
              "borderLeftWidth": 2,
              "borderRightWidth": 2,
              "borderTopWidth": 2,
            },
            {
              "borderBottomColor": "#6bc0eb",
              "borderLeftColor": "#6bc0eb",
              "borderRightColor": "#6bc0eb",
              "borderTopColor": "#6bc0eb",
            },
            {
              "borderBottomLeftRadius": 24,
              "borderBottomRightRadius": 24,
              "borderTopLeftRadius": 24,
              "borderTopRightRadius": 24,
            },
          ]
        }
        testID="grazing-button"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Nunito-SemiBold",
              },
              {
                "color": "#ffffff",
              },
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
              {
                "textAlign": "center",
              },
            ]
          }
        >
          Ver pastoreo
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 8,
          "marginRight": 8,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "display": "flex",
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "backgroundColor": "#272727",
          },
          {
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
          },
        ]
      }
    >
      <View>
        <View
          style={
            [
              {
                "marginTop": -4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontFamily": "Nunito-Light",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            Grupos
            :
          </Text>
          <View
            style={
              [
                {
                  "backgroundColor": "#ffffff",
                },
                {
                  "borderBottomWidth": 2,
                  "borderLeftWidth": 2,
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                },
                {
                  "borderBottomColor": "#2254ba",
                  "borderLeftColor": "#2254ba",
                  "borderRightColor": "#2254ba",
                  "borderTopColor": "#2254ba",
                },
                {
                  "borderBottomLeftRadius": 24,
                  "borderBottomRightRadius": 24,
                  "borderTopLeftRadius": 24,
                  "borderTopRightRadius": 24,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            <View>
              <RNCPicker
                items={[]}
                numberOfLines={1}
                onChange={[Function]}
                selectedIndex={0}
                style={
                  [
                    {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
                testID="group-picker"
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "marginTop": 24,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "marginTop": -4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontFamily": "Nunito-Light",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            Fincas
            :
          </Text>
          <View
            style={
              [
                {
                  "backgroundColor": "#ffffff",
                },
                {
                  "borderBottomWidth": 2,
                  "borderLeftWidth": 2,
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                },
                {
                  "borderBottomColor": "#2254ba",
                  "borderLeftColor": "#2254ba",
                  "borderRightColor": "#2254ba",
                  "borderTopColor": "#2254ba",
                },
                {
                  "borderBottomLeftRadius": 24,
                  "borderBottomRightRadius": 24,
                  "borderTopLeftRadius": 24,
                  "borderTopRightRadius": 24,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            <View>
              <RNCPicker
                items={[]}
                numberOfLines={1}
                onChange={[Function]}
                selectedIndex={0}
                style={
                  [
                    {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
                testID="field-picker"
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "marginBottom": 32,
              "marginTop": 32,
            },
            [
              {
                "marginTop": 24,
              },
            ],
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "display": "flex",
              },
              {
                "flexDirection": "row",
              },
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
              {
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              },
              {
                "backgroundColor": "#272727",
              },
              {
                "borderBottomWidth": 2,
                "borderLeftWidth": 2,
                "borderRightWidth": 2,
                "borderTopWidth": 2,
              },
              {
                "borderBottomColor": "#6bc0eb",
                "borderLeftColor": "#6bc0eb",
                "borderRightColor": "#6bc0eb",
                "borderTopColor": "#6bc0eb",
              },
              {
                "borderBottomLeftRadius": 24,
                "borderBottomRightRadius": 24,
                "borderTopLeftRadius": 24,
                "borderTopRightRadius": 24,
              },
            ]
          }
          testID="handle-add-grazing-button"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Nunito-SemiBold",
                },
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
                [
                  {
                    "marginLeft": 16,
                  },
                ],
              ]
            }
          >
            Añadir
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
